src/dotnet/ShellShim/IShellShimRepository.cs

```diff
@@ -6,8 +6,7 @@ namespace Microsoft.DotNet.ShellShim
 {
     internal interface IShellShimRepository
     {
-        void CreateShim(FilePath targetExecutablePath, string commandName);
-
+        void CreateShim(FilePath targetExecutablePath, string commandName, DirectoryPath? nonGlobalLocation = null);
```

Comment:
This seems like the wrong abstraction to me as it makes "global" vs. "non-global" a concern of the caller.  I'd expect both the shim respository and the package store to simply be created with different paths when doing a "non-global" install.

Reply:
I tried that approve in the beginning. And then I realize it is the concern of the caller. When user type `install --bin-path` this is an explicit concern that the caller have to real with,

I was thinking about the caller create a shellshimrepositoy during the call, but that is worse. Creating a ShellshimRepository is not the responsibility of the caller